wallet/wallet-node-client/src/rpc_client/dummy_client.rs

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate::ChainInfo;
+use common::{
+    chain::{
+        tokens::{RPCTokenInfo, TokenId},
+        Block, DelegationId, GenBlock, PoolId, SignedTransaction, Transaction,
+    },
+    primitives::{Amount, BlockHeight, Id},
+};
+use consensus::GenerateBlockInputData;
+use crypto::ephemeral_e2e::EndToEndPublicKey;
+use mempool::{tx_accumulator::PackingStrategy, tx_options::TxOptionsOverrides, FeeRate};
+use p2p::{
+    interface::types::ConnectedPeer,
+    types::{bannable_address::BannableAddress, ip_or_socket_address::IpOrSocketAddress, PeerId},
+};
+
+use crate::node_traits::NodeInterface;
+
+use super::{MaybeDummyNode, NodeRpcError};
+
+#[async_trait::async_trait]
+impl NodeInterface for MaybeDummyNode {
+    type Error = NodeRpcError;
+
+    async fn chainstate_info(&self) -> Result<ChainInfo, Self::Error> {
+        if let Some(client) = self.opt_client.as_ref() {
+            client.chainstate_info().await
+        } else {
+            let genesis = self.chain_config.genesis_block();
+            Ok(ChainInfo {
+                best_block_id: self.chain_config.genesis_block_id(),
+                best_block_height: BlockHeight::zero(),
+                best_block_timestamp: genesis.timestamp(),
+                median_time: genesis.timestamp(),
+                is_initial_block_download: false,
+            })
+        }
+    }
+
+    async fn get_block(&self, block_id: Id<Block>) -> Result<Option<Block>, Self::Error> {
+        if let Some(client) = self.opt_client.as_ref() {
+            client.get_block(block_id).await
+        } else {
+            Err(NodeRpcError::ResponseError(
+                jsonrpsee::core::ClientError::HttpNotImplemented,
+            ))
+        }
+    }
+
+    async fn get_mainchain_blocks(
+        &self,
+        from: BlockHeight,
+        max_count: usize,
+    ) -> Result<Vec<Block>, Self::Error> {
+        if let Some(client) = self.opt_client.as_ref() {
+            client.get_mainchain_blocks(from, max_count).await
+        } else {
+            Err(NodeRpcError::ResponseError(
```

Comment:
Also could we replace this with a more meaningful error? Remember that users will be using this and will be getting "not implemented". This isn't really helpful. Please create a special type of errors for the cold wallet, and explain in it what the error means. Something like "This command requires communication with the node, but this cannot happen in cold-wallet mode". Most errors will be that one, but if need be others, please explain it.

Reply:
The unsupported commands are filtered out before reaching here, so this error will never trigger

Reply:
Still, this is not 100% filtered in RPC. So let's have some good error support so that if something goes wrong, users would understand.